src/hardware/nuvoton-M051/cpu/CMSIS/Device/Nuvoton/M051Series/Source/GCC/gcc_arm.ld

```diff
@@ -44,6 +45,11 @@ ENTRY(Reset_Handler)
 
 SECTIONS
 {
+  .data_flash (NOLOAD):
+  {
+      *(.data_flash*)

```

Comment:
where is this part from?
I'm wondering if it's needed?
what happens when the part is missing?  (code is compiling).

I'm also not sure if:
```
DATA_FLASH (rx) : ORIGIN = 0x0001F000, LENGTH = 1000 /*  4k */
```
the `LENGTH` is correct?

Reply:
That part is necessary to access the Nuvoton’s persistent RW memory (which their datasheets call the “data flash”). Its used in place of an EEPROM in cheali. The length comes from the datasheet which states the section is 4k (or hex 0x1000). The code will compile for the charger without it, but it will fail to boot because the EEPROM checks will fall

Reply:
Documentation:
NUC029LAN - page 242; https://www.nuvoton.com/export/resource-files/TRM_NUC029xAN_xAE_Series_EN_Rev1.09.pdf
M0516LDN - page 146; https://www.nuvoton.com/export/resource-files/TRM_M051%28BN_DN_DE%29_Series_EN_Rev1.03.pdf

Reply:
Lengths should be fixed in the latest commit